### PR TITLE
FEA: add SimpleX in general model

### PR DESCRIPTION
**Pull request by dreaming-qin**
add SimpleX in general model

### CODE REVIEW

**Review by Wicknight — APPROVED** [review]

**Comment by Wicknight** [comment]
@dreaming-qin Hello! Thanks for your attention to RecBole!
We have checked the code you submitted and found no problems. After that, we will conduct internal tests on various datasets. When the tests are completed, we will merge the code immediately. You are welcome to submit more codes to communicate with us.

File: recbole/model/general_recommender/simplex.py [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+
+    We implement the model following the original author with a pairwise training mode.
+    """
+    input_type = InputType.PAIRWISE
+
+    def __init__(self, config, dataset):
+        super(SimpleX, self).__init__(config, dataset)
+
+        # Get user transaction history
+        self.history_item_id, _, self.history_item_len = dataset.history_item_matrix()
```
Comment (Wicknight) [inline recbole/model/general_recommender/simplex.py:43]: 在gpu运行时会报错，需要在这里添加self.history_item_id = self.history_item_id.to(self.device)以及self.history_item_len=self.history_item_len.to(self.device)

File: recbole/model/general_recommender/simplex.py [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+        # user's id
+        user = user[0:user_number]
+        # historical transaction record
+        history_item = self.history_item_id[user]
+        history_item = history_item[:, :self.history_len]
+        # positive item's id
+        pos_item = pos_item[0:user_number]
+        # history_len
+        history_len = self.history_item_len[user]
+        history_len = torch.minimum(history_len, torch.zeros(1)+self.history_len)
```
Comment (Wicknight) [inline recbole/model/general_recommender/simplex.py:204]: history_len=torch.minimum(history_len, torch.zeros(1, device=self.device)+self.history_len)

File: recbole/model/general_recommender/simplex.py [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+
+        loss = self.forward(user, pos_item, history_item, history_len, neg_item_seq)
+        return loss
+
+    def predict(self, interaction):
+        user = interaction[self.USER_ID]
+        item_seq = self.history_item_id[user]
+        item_seq = item_seq[:, :self.history_len]
+        item_seq_len = self.history_item_len[user]
+        item_seq_len = torch.minimum(item_seq_len, torch.zeros(1)+self.history_len)
```
Comment (Wicknight) [inline recbole/model/general_recommender/simplex.py:214]: item_seq_len=torch.minimum(item_seq_len, torch.zeros(1, device=self.device)+self.history_len)

File: recbole/model/general_recommender/simplex.py [hunk 4]
```diff
@@ -0,0 +1,251 @@
...
+
+        UI_cos = self.get_cos(UI_aggregation_e, test_item_e.unsqueeze(1))
+        return UI_cos.squeeze()
+
+    def full_sort_predict(self, interaction):
+        user = interaction[self.USER_ID]
+        item_seq = self.history_item_id[user]
+        item_seq = item_seq[:, :self.history_len]
+        item_seq_len = self.history_item_len[user]
+        item_seq_len = torch.minimum(item_seq_len, torch.zeros(1)+self.history_len)
```
Comment (Wicknight) [inline recbole/model/general_recommender/simplex.py:236]: item_seq_len=torch.minimum(item_seq_len, torch.zeros(1, device=self.device)+self.history_len)

**Review by Wicknight — CHANGES_REQUESTED** [review]

File: recbole/model/general_recommender/simplex.py (hunk 4 above)
Comment (dreaming-qin) [reply recbole/model/general_recommender/simplex.py]: ok，已进行修改并已提交。除此之外还修改了predict方法与full_sort_predict方法的部分变量名称让变量更通俗易懂

**Review by Wicknight — APPROVED** [review]

**Pull request merged**